crates/rome_js_parser/src/syntax/expr.rs

```diff
@@ -1811,15 +1811,27 @@ pub(super) fn parse_unary_expr(p: &mut Parser, context: ExpressionContext) -> Pa
         // // SCRIPT
         // async function test() {}
         // await test();
+        let checkpoint = p.checkpoint();
         let m = p.start();
```

Comment:
```suggestion
        let m = p.start();
        let checkpoint = p.checkpoint();
```

Nit: You can move the `checkpoint` after `let m = p.start()` since you are calling `m.abandon()` anyway (I'm somewhat surprised that it isn't throwing because `rewind` already deletes the marker.

Reply:
Updated